metricflow/dataset/sql_dataset.py

```diff
@@ -153,6 +154,14 @@ def instance_for_time_dimension(self, time_dimension_spec: TimeDimensionSpec) ->
             )
         return instances[0]
 
+    def instance_for_spec(self, spec: InstanceSpec) -> MdoInstance:
+        """Given a spec, return the instance associated with it in the data set."""
+        instances = self.instance_set.as_tuple
+        for instance in instances:
+            if instance.spec == spec:
+                return instance
+        raise RuntimeError(f"Did not find instance matching spec in dataset. Spec: {spec}\nInstances: {instances}")
```

Comment:
For better formatting, you can do:

```
# Easier autocomplete.
raise RuntimeError(str(LazyFormat("Did not find instance matching spec in dataset.", spec=spec, instances=instances))

# Or call formatting function.
raise RuntimeError(mf_pformat_dict("Did not find instance matching spec in dataset.", {"spec": spec, "instances": instances))
```

Reply:
Ah good call! I need to change this habit.